docs/common/project/how-to-release.md

```diff
@@ -255,3 +260,60 @@ Java artifacts are available from Maven Central.
 
 Thanks to everyone for contributing!
 ```
+
+### Documentation Release
+
+Documentation and website needs to be updated as a part of Iceberg release.
+
+#### Iceberg version update
+
+Before release, publish a PR to update `landing-page/config.toml` and `docs/config.toml` with the new Iceberg version number.
+
+#### Documentation update
+
+To start the release process, run the following steps in the `iceberg-docs` repository to copy docs over 
+(assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally, and you are in the `iceberg` repository):
+
+```shell
+rm -rf ../iceberg-docs/docs/content/docs
+rm -rf ../iceberg-docs/landing-page/content/common
+cp -r docs/versioned ../iceberg-docs/docs/content/docs
+cp -r docs/common ../iceberg-docs/landing-page/content/common
+```
+
+This should be approved in a separated PR.
+
+#### Javadoc update
+
+In `iceberg` repository, generate the javadoc for your release and copy to the `javadoc` folder in `iceberg-docs` repo 
+(assume the `iceberg-docs` repository and `iceberg` repository are in the same parent directory locally, and you are in the `iceberg` repository):
+
+```shell
+./gradlew refreshJavadoc
+rm -rf ../iceberg-docs/javadoc
+cp site/docs/javadoc/<VERSION NUMBER> ../iceberg-docs/javadoc
+```
+
+This should be approved in a separated PR.
+
+### Cut new version branch
+
+Once completed, go to the `iceberg-docs` repository (the Apache one, not a forked one) to update branches.
```

Comment:
`update branches` isn't clear here. Do you mean update existing branches for previous releases? Isn't that only needed if there's a hot-fix for a previous release?

Reply:
updated, let me know if the new wording is better